Minor code refactoring and makefile fix

nmnobre opened the pull request:
Refs #31217

I don't have an arm Mac, so I'll be hammering CIVET to reverse engineer this one...

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3100076/), step Generate coverage on 38a375e wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31221/coverage/xfem)

</details>



This comment will be updated on new commits.

nmnobre committed:
Fix makefile deps by setting same all_header_dir in prerequisites and in recipe (#31221)

A completely fresh build (i.e. missing framework/include/base/MooseRevision.h)
would _always_ fail with make -j1. This _might_ have also been a cause of
intermittent failures when building in parallel.

nmnobre committed:
Refactor SteadyBase's & MFEMSteady's ctors so the latter matches MFEMTransient's (#31221)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3114898/), step Versioner verify on 2541fe7 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 1 changed, 0 added, 0 removed

| package   | status   | file                               |
|-----------|----------|------------------------------------|
| moose-dev | CHANGE   | scripts/update_and_rebuild_mfem.sh |

### Versioner versions

Found 9 packages, 1 changed, 1 failed

| package   | status    | hash    | to hash   | version    | to version   |
|-----------|-----------|---------|-----------|------------|--------------|
| moose-dev | NEED BUMP | 46da177 | 0a2c9c5   | 2025.07.22 | 2025.07.22   |

Verification failed.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3114965/), step Versioner verify on 46e54c2 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 1 changed, 0 added, 0 removed

| package   | status   | file                               |
|-----------|----------|------------------------------------|
| moose-dev | CHANGE   | scripts/update_and_rebuild_mfem.sh |

### Versioner versions

Found 9 packages, 1 changed, 1 failed

| package   | status    | hash    | to hash   | version    | to version   |
|-----------|-----------|---------|-----------|------------|--------------|
| moose-dev | NEED BUMP | 46da177 | 0a2c9c5   | 2025.07.22 | 2025.07.22   |

Verification failed.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3117297/), step Versioner verify on 9345751 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 3 changed, 0 added, 0 removed

| package   | status   | file                               |
|-----------|----------|------------------------------------|
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml          |
| moose-dev | CHANGE   | framework/contrib/mfem             |
| moose-dev | CHANGE   | scripts/update_and_rebuild_mfem.sh |

### Versioner versions

Found 9 packages, 1 changed, 0 failed

| package   | status   | hash    | to hash   | version    | to version   |
|-----------|----------|---------|-----------|------------|--------------|
| moose-dev | CHANGE   | 46da177 | a7efe75   | 2025.07.22 | 2025.08.11   |

loganharbour commented:
Would like to merge this (the update part) into https://github.com/idaholab/moose/pull/31141 if possible. We'll see

nmnobre committed:
Remove unnecessary code in TimeDomainProblemOperator (#31221)

nmnobre committed:
Remove host read previously required by conduit data outputs (#31221)

lindsayad commented:
Does this include https://github.com/mfem/mfem/pull/4923? I would like to make sure it's included

lindsayad reviewed (approved):
this LGTM once the update PR is merged

nmnobre committed:
Remove unnecessary conversion between std::vector/std::set (#31221)

lindsayad reviewed (commented):
I'm unsubscribing after granting my approval. Mention me if you need me for anything or if you add something significant on top of what you've already done

nmnobre committed:
Remove unnecessary max_parallel requirement (#31221)

nmnobre committed:
Fix MFEMVectorIC's vector coefficient name (#31221)

nmnobre committed:
Simplify diffusion test input specification (#31221)

nmnobre committed:
Fix MFEMMesh docs (#31221)

nmnobre commented:
> Mention me [...] if you add something significant on top of what you've already done

I don't know if it's significant, but it's stuff you once added that I just removed from `diffusion.i`. That are other tests making (more meaningful) use of the functionality showcased in the removed lines and I've been asking people to expand on `diffusion.i` instead of creating new test files, so it's just to keep the no. of lines under control.
